Source/msg.cpp

```diff
@@ -355,7 +355,8 @@ const byte *DeltaImportObjects(const byte *src, std::unordered_map<WorldTilePosi
 	dst.reserve(numDeltas);
 
 	for (unsigned i = 0; i < numDeltas; i++) {
-		WorldTilePosition objectPosition { static_cast<WorldTileCoord>(*src++), static_cast<WorldTileCoord>(*src++) };
+		WorldTilePosition objectPosition { static_cast<WorldTileCoord>(src[0]), static_cast<WorldTileCoord>(src[1]) };
+		src += 2;

```

Comment:
This was/is using [braced list initialisation](https://en.cppreference.com/w/cpp/language/list_initialization).

> Every initializer clause is [sequenced before](https://en.cppreference.com/w/cpp/language/eval_order) any initializer clause that follows it in the braced-init-list.

This should guarantee that all side-effects of the first post-increment are evaluated (and stored) before the second post-increment is executed.

Reply:
I guess I learn something new every day. Cheers for that!